zen/screen/zigzag-layout.c

```diff
@@ -0,0 +1,280 @@
+#include "zen/screen/zigzag-layout.h"
+
+#include <cairo-ft.h>
+#include <librsvg/rsvg.h>
+#include <time.h>
+#include <wlr/types/wlr_output_damage.h>
+#include <zen-common.h>
+#include <zigzag.h>
+
+#include "zen/server.h"
+
+const int menu_bar_height = 33.;
+
+struct zn_zigzag_layout_state {
+  struct zn_output *output;
+  struct zigzag_node *power_button;
+  struct wl_event_source *minute_timer_source;
+  long next_min_ms;
+};
+
+static int
+zn_output_handle_minute_timer(void *data)
+{
+  struct zigzag_layout *self = (struct zigzag_layout *)data;
+  struct zn_zigzag_layout_state *state =
+      (struct zn_zigzag_layout_state *)self->state;
+  struct zn_server *server = zn_server_get_singleton();
+  long time_ms = current_time_ms();
```

Comment:
`current_time_ms` uses REALTIME CLOCK with which the clock can change significantly when, for example, NTP synchronization. We should use `clock_gettime(CLOCK_MONOTONIC, &ts);` in this case.

```suggestion
  struct timespec now;
  clock_gettime(CLOCK_MONOTONIC, &now);
  int64_t now_ms = timespec_to_msec(&now);
```

The same applies to the other use of `current_time_ms`.

Reply:
The "system clock" here does not show the time elapsed in the process but shows the real time clock, so the timer update should happen based on the `CLOCK_REALTIME` in that sense, the time going back and forth on NTP sync is completely reasonable. I keep using `CLOCK_REALTIME`, but changed the interval of the timer to a second so that the time change based on the NTP clock happens immediately. 

I also made a change to re-calculate the time the timer is fired in every handler call so that `CLOCK_REALTIME` change does not cause a cumulative problem.